src/actions/textobject.ts

```diff
@@ -640,3 +641,219 @@ class InsideIndentObjectBoth extends IndentObjectMatch {
   includeLineAbove = true;
   includeLineBelow = true;
 }
+
+abstract class SelectArgument extends TextObjectMovement {
+  modes = [Mode.Normal, Mode.Visual];
+
+  private static openingDelimiterCharacters(): string[] {
+    return configuration.argumentObjectOpeningDelimiters;
+  }
+  private static closingDelimiterCharacters(): string[] {
+    return configuration.argumentObjectClosingDelimiters;
+  }
+  private static separatorCharacters(): string[] {
+    return configuration.argumentObjectSeparators;
+  }
+
+  protected selectAround = false;
+
+  // Requirement is that below example still works as expected, i.e.
+  // when we have nested pairs of parens
+  //
+  //        ( a, b, (void*) | c(void*, void*), a)
+  //
+  // Warning: For now, mismatched opening and closing delimiters, e.g.
+  // in (foo] will still be matched by this movement.
+  //
+  // Procedure:
+  //
+  // 1   Find boundaries left/right (i.e. where the argument starts/ends)
+  // 1.1 Walk left until we find a comma or an opening paren, that does not
+  //     have a matching closed one. This way we can ignore pairs
+  //     of parentheses which are part of the current argument.
+  // 1.2 Vice versa for walking right.
+  // 2   Depending on our mode (inner or around), improve the start/stop
+  //     locations for most consistent behaviour, especially in case of
+  //     multi-line statements.
+
+  public async execAction(position: Position, vimState: VimState): Promise<IMovement> {
+    const failure = { start: position, stop: position, failed: true };
+    let leftSearchStartPosition = position;
+    let rightSearchStartPosition = position;
+
+    // When the cursor is on a delimiter already, pre-advance the cursor,
+    // so that our search actually spans a range. We will advance to the next argument,
+    // in case of opening delimiters or separators, and advance to the
+    // previous on closing delimiters.
+    if (
+      SelectArgument.separatorCharacters().includes(TextEditor.getCharAt(position)) ||
+      SelectArgument.openingDelimiterCharacters().includes(TextEditor.getCharAt(position))
+    ) {
+      rightSearchStartPosition = position.getRightThroughLineBreaks(true);
+    } else if (
+      SelectArgument.closingDelimiterCharacters().includes(TextEditor.getCharAt(position))
+    ) {
+      leftSearchStartPosition = position.getLeftThroughLineBreaks(true);
+    }
+
+    const leftArgumentBoundary = SelectInnerArgument.findLeftArgumentBoundary(
+      leftSearchStartPosition
+    );
+    const rightArgumentBoundary = SelectInnerArgument.findRightArgumentBoundary(
+      rightSearchStartPosition
+    );
+
+    if (leftArgumentBoundary === undefined || rightArgumentBoundary === undefined) {
```

Comment:
Check `leftArgumentBoundary === undefined` before you calculate `rightArgumentBoundary` to avoid unnecessary computation